src/main/scala/com/fulcrumgenomics/bam/ComputeRnaSeqInsertSize.scala

```diff
@@ -0,0 +1,330 @@
+ @arg(flag="r", doc="Input gene annotations in refFlat form.  Format described here: http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat")
+ val refFlat: FilePath,
+ @arg(flag="o", doc="Output metrics file to which to write.") val output: FilePath
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertReadable(refFlat)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    val progress = new ProgressLogger(logger, verb = "read", unit = 5e6.toInt)
+    val in = SamReaderFactory.make.open(input)
+    val refFlatSource = RefFlatSource(refFlat)
+    val geneOverlapDetector = new OverlapDetector[Gene](0, 0)
+    refFlatSource.foreach { gene => geneOverlapDetector.addLhs(gene, gene) }
+    val counters = PairOrientation.values().toList.map { pairOrientation => (pairOrientation, new NumericCounter[Long]()) }.toMap
+
+    in.foreach { rec =>
+      calculateInsertSize(rec=rec, overlapDetector=geneOverlapDetector) match {
+        case None             => Unit // ignore
+        case Some(insertSize) => counters(SamPairUtil.getPairOrientation(rec)).count(insertSize)
+      }
+      progress.record(rec)
+    }
+
+    refFlatSource.safelyClose()
+    in.safelyClose()
+
+    val metrics = PairOrientation.values().toList.map { pairOrientation =>
+      val counter = counters(pairOrientation)
+      val mean    = counter.mean()
+      InsertSizeMetric(
+        pairOrientation    = pairOrientation,
+        read_pairs         = counter.total,
+        mean               = mean,
+        median             = counter.median(),
+        min                = counter.map(_._2).min,
+        max                = counter.map(_._2).max,
+        standard_deviation = counter.stddev(m=mean)
+      )
+    }
+
+    Metric.write(metrics=metrics, path=output)
+  }
+
+  /** Calculates the insert size in transcript space. */
+  private def calculateInsertSize(rec: SAMRecord, overlapDetector: OverlapDetector[Gene]): Option[Int] = {
+    if (!rec.getReadPairedFlag || rec.getReadFailsVendorQualityCheckFlag || rec.isSecondaryOrSupplementary) return None
+    else if (rec.getReadUnmappedFlag || rec.getMateUnmappedFlag) return None
+    else if (rec.getDuplicateReadFlag || rec.getReferenceIndex != rec.getMateReferenceIndex) return None
+    else if (rec.getSecondOfPairFlag) return None
+
+    val mateCigar        = SAMUtils.getMateCigar(rec)
+    require(mateCigar != null, s"Mate CIGAR (Tag MC) not found for $rec.")
+    val mateAlignmentEnd = CoordMath.getEnd(rec.getMateAlignmentStart, mateCigar.getReferenceLength)
+
+    // Check for overlapping reads
+    if (CoordMath.overlaps(rec.getAlignmentStart, rec.getAlignmentEnd, rec.getMateAlignmentStart, mateAlignmentEnd)) {
+      fail(s"Record overlaps with its mate, consider using ClipOverlappingReads.  Record: $rec")
+    }
+
+    // Get the overlapping genes
+    val leftMostBase     = Math.min(rec.getAlignmentStart, rec.getMateAlignmentStart)
```

Comment:
I think we should discuss what the right pair of coordinates to use are.  My initial reaction is that I'd prefer to stick with the Picard/htsjdk style of using the 5' coordinate of each read, and the distance between them as the insert size.  My take is that:

1. Definitely makes the most sense for `FR` reads
2. Is the most predictable/interpretable thing to do for `RF` reads
3. Makes as much sense as anything else for `TANDEM` reads
4. Side-steps any issues with improperly clipped ends of reads